pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -650,3 +761,43 @@ func osdOnSDNFlag(hostnetwork bool, v cephver.CephVersion) []string {
 
 	return args
 }
+
+func makeStorageClassDeviceSetPVC(storageClassDeviceSetName, pvcStorageClassDeviceSetPVCId string, pvcIndex, setIndex int, pvcTemplate v1.PersistentVolumeClaim) (pvcs *v1.PersistentVolumeClaim) {
+	pvcLabels := makeStorageClassDeviceSetPVCLabel(storageClassDeviceSetName, pvcStorageClassDeviceSetPVCId, pvcIndex, setIndex)
+
+	// pvc naming format <storageClassDeviceSetName>-<SetNumber>-<PVCIndex>
+	pvcGenerateName := pvcStorageClassDeviceSetPVCId + "-"
+
+	// Add pvcTemplate name to pvc name. i.e <storageClassDeviceSetName>-<SetNumber>-<PVCIndex>-<pvcTemplateName>
+	if len(pvcTemplate.GetName()) != 0 {
+		pvcGenerateName = pvcGenerateName + pvcTemplate.GetName() + "-"
+	}
+
+	// Add user provided labels to pvcTemplates
+	for k, v := range pvcTemplate.GetLabels() {
+		pvcLabels[k] = v
+	}
+
+	return &v1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: pvcGenerateName,
+			Labels:       pvcLabels,
+			Annotations:  pvcTemplate.Annotations,
+		},
+		Spec: pvcTemplate.Spec,
+	}
+}
+
+func makeStorageClassDeviceSetPVCID(storageClassDeviceSetName string, setIndex, pvcIndex int) (pvcId, pvcLabelSelector string) {
+	pvcStorageClassDeviceSetPVCId := fmt.Sprintf("%s-%v-%v", storageClassDeviceSetName, setIndex, pvcIndex)
+	return pvcStorageClassDeviceSetPVCId, fmt.Sprintf("ceph.rook.io/StorageClassDeviceSetPVCId=%s", pvcStorageClassDeviceSetPVCId)
```

Comment:
StorageClassDeviceSet is such a long name to expose. How about in places like this we use a nickname of DeviceSet, which would mean here we use `DeviceSetPVCId`?

Reply:
cc @ashishranjan738